tests/integration/test_structured_logging_json/test.py

```diff
@@ -85,9 +95,12 @@ def validate_log_config_relation(config, logs, config_type):
                 keys_in_log.add(log_key)
                 if log_key not in keys_in_config:
                     return False
-            for config_key in keys_in_config:
-                if config_key not in keys_in_log:
-                    return False
```

Comment:
As long as the present keys in the logs are valid, the restriction could be lifted.
Some keys are not always relevant for every log entry, enforcing its presence in every entry could be too restrictive.
But if this is strictly required that all log entries must include all keys, this field could be added again.

Reply:
Full disclaimer: I'm not a ClickHouse maintainer, just a bystander who happens to also want this feature shipped :)

The reason I asked the question is because I expect this assertion logic exists for a specific reason. Does this cause problems / conflict with the changes you made?

Reply:
Yeah I know :)
All the points you made so far were valid so I appreciate your opinion.

I do not think that this would case any problems, in fact this approach is more efficient by not forcing all keys to exists (even when the keys in logs are not relevant to the job)